database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowStoreEntryComponent.scala

```diff
@@ -73,31 +73,96 @@ trait WorkflowStoreEntryComponent {
   )
 
   /**
-    * Returns up to "limit" startable workflows, sorted by submission time.
-    */
-  def fetchStartableWorkflows(limit: Long,
-     heartbeatTimestampTimedOut: Timestamp,
-     excludeWorkflowState: String,
-     excludedGroups: Set[String]
-    ): Query[WorkflowStoreEntries, WorkflowStoreEntry, Seq] = {
-      val query = for {
-        row <- workflowStoreEntries
-        /*
+   * Return hog group with the lowest count of actively running workflows
+   */
+  def getHogGroupWithLowestRunningWfs(heartbeatTimestampTimedOut: Timestamp,
+                                      excludeWorkflowState: String,
+                                      excludedGroups: Set[String]): Query[Rep[Option[String]], Option[String], Seq] = {
+    val startableWorkflows = for {
+      row <- workflowStoreEntries
+      /*
         This looks for:
-
         1) Workflows with no heartbeat (newly submitted or from a cleanly shut down Cromwell).
         2) Workflows with old heartbeats, presumably abandoned by a defunct Cromwell.
+        3) Workflows not in "OnHold" state
+        4) Workflows that don't belong to hog groups in excludedGroups
+      */
+      if (row.heartbeatTimestamp.isEmpty || row.heartbeatTimestamp < heartbeatTimestampTimedOut) &&
+        (row.workflowState =!= excludeWorkflowState) &&
+        !(row.hogGroup inSet excludedGroups)
+    } yield row
+
+    // calculates the count of startable workflows per hog group and oldest submission timestamp of workflow
+    // present in that hog group
+    val numOfStartableWfsByHogGroup = startableWorkflows
```

Comment:
These `val`s might benefit from type annotations

Reply:
I had originally removed them because the type annotation was quite long. I can add them back in. What is usually the standard when type annotations are very long? Are they split into multiple lines? Or should they be kept on same line? For example, here with type annotation it would look like:
```
val numOfStartableWfsByHogGroup: Query[(Rep[Option[String]], Rep[Int], Rep[Option[Timestamp]]), (Option[String], Int, Option[Timestamp]), Seq] = startableWorkflows
      .groupBy(_.hogGroup)
      .map { case (hogGroupName, workflows) => (hogGroupName, workflows.length, workflows.map(_.submissionTime).min) }
      .sortBy(_._2.asc)
```

Reply:
Oh yeah haha, I added the type annotation while reviewing the branch locally in IntelliJ and then I was like "nope nope nope"

Reply:
@cjllanwarne I have added type annotations for few variables in this function and split them on multiple lines for readability. Let me know if that makes understanding the code easier. If splitting on multiple lines feels clunky I will revert the changes.